packages/zoe/tools/manualTimer.js

```diff
@@ -1,37 +1,79 @@
+// @ts-check
+
 import { E } from '@agoric/eventual-send';
+import { makeStore } from '@agoric/store';
+
+import './types';
+
+/* eslint-disable jsdoc/valid-types */
+/**
+ * @typedef {Object} ManualTimerAdmin
+ * @property {(msg?: string) => Promise<void>} tick
+ */
+/* eslint-enable jsdoc/valid-types */
 
-// A fake clock that also logs progress in tests.
+/**
+ * @typedef {ManualTimerAdmin & TimerService} ManualTimer
+ */
+
+/**
+ * A fake clock that also logs progress.
+ *
+ * @param {(...args: any[]) => void} log
+ * @param {Timestamp} [startValue=0]
+ * @returns {ManualTimer}
+ */
 export default function buildManualTimer(log, startValue = 0) {
   let ticks = startValue;
-  const schedule = new Map();
+
+  /** @type {Store<Timestamp, Array<TimerServiceHandler>>} */
+  const schedule = makeStore('Timestamp');
+
+  /** @type {ManualTimer} */
   const timer = {
-    setWakeup(deadline, handler) {
-      if (deadline <= ticks) {
-        log(`&& task was past its deadline when scheduled: ${deadline} &&`);
-        handler.wake(ticks);
-        return undefined;
-      }
-      log(`@@ schedule task for:${deadline}, currently: ${ticks} @@`);
-      if (!schedule.has(deadline)) {
-        schedule.set(deadline, []);
-      }
-      schedule.get(deadline).push(handler);
-      return deadline;
-    },
     // This function will only be called in testing code to advance the clock.
     async tick(msg) {
       ticks += 1;
       log(`@@ tick:${ticks}${msg ? `: ${msg}` : ''} @@`);
       if (schedule.has(ticks)) {
+        const handlers = schedule.get(ticks);
```

Comment:
`Handler` to me means a function, but these are objects due to our marshalling restrictions. Can we call them `handlerObj` or something?

Reply:
How about `Waker`?

Reply:
Waker it is!